resources/functionMetadata.php

```diff
@@ -811,47 +811,47 @@
 	'ctype_upper' => ['hasSideEffects' => false],
 	'ctype_xdigit' => ['hasSideEffects' => false],
 	'curl_copy_handle' => ['hasSideEffects' => false],
-	'curl_errno' => ['hasSideEffects' => false],
-	'curl_error' => ['hasSideEffects' => false],
+	'curl_errno' => ['hasSideEffects' => true],
+	'curl_error' => ['hasSideEffects' => true],
```

Comment:
This also looks wrong. I guess you should review all lines which changed from false to true

Reply:
Although they appear to be pure, they actually rely on some external state associated with the resource, so it makes sense that PhpStorm marks them as `#[Pure(true)]`.

While it may be convenient to make `curl_errno()` pure for convenience in common use cases, it is inconsistent with impure functions like `ob_get_level()`(#12577).

To solve this problem, [`must_use`/`#[NoDiscard]`](https://github.com/phpstan/phpstan/issues/12738) needs to be supported separately from `@pure`/`@impure`.